core/jvm/src/test/scala/scalaz/zio/SemaphoreSpec.scala

```diff
@@ -43,6 +44,15 @@ class SemaphoreSpec extends AbstractRTSSpec {
       (s, permits) => IO.parTraverse(permits.reverse)(amount => s.releaseN(amount)).void
     )
 
+  def e5 = {
+    val n = 1L
+    unsafeRun(for {
+      s <- Semaphore(n).peek(_.acquire)
+      _ <- s.release.fork
+      _ <- s.acquire
+    } yield () must_=== (()))
+  }
+
```

Comment:
We may want to add one crazy test in here that does stuff from a lot of fibers. The above one is certainly helpful.